Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -92,7 +92,28 @@ namespace AzToolsFramework
 
         AZStd::unique_ptr<Instance> PrefabSystemComponent::CreatePrefab(
             const AZStd::vector<AZ::Entity*>& entities, AZStd::vector<AZStd::unique_ptr<Instance>>&& instancesToConsume,
-            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity, bool shouldCreateLinks)
+            AZ::IO::PathView filePath, AZStd::unique_ptr<AZ::Entity> containerEntity, InstanceOptionalReference parent,
+            bool shouldCreateLinks)
+        {
+
+            AZStd::unique_ptr<Instance> newInstance;
+
+            if (parent.has_value())
+            {
+                newInstance = AZStd::make_unique<Instance>(parent);
+            }
+            else
+            {
+                newInstance = AZStd::make_unique<Instance>(AZStd::move(containerEntity));
+            }
```

Comment:
Nitpick: very minor, but since you were using ternary operators early, for consistency you can also the follow I think:
```
auto newInstance = AZStd::make_unique<Instance>(parent.has_value() ? parent : AZStd::move(containerEntity));
```